utility/auth.js

```diff
@@ -0,0 +1,72 @@
+import React, { createContext, useState, useContext, useEffect } from 'react'
+import { useRouter } from 'next/router'
+import firebase from 'firebase'
+import styled from 'styled-components'
+
+const LoadingScreenContainer = styled.div`
+    display: flex;
+    align-items: center;
+    justify-content: center;
+    height: 100vh;
+`
+
+const LoadingDiv = styled.div`
+    font-size: 50px;
+`
+
+export const checkAdminClaim = async user => {
+    const token = await user.getIdTokenResult()
+    return Object.prototype.hasOwnProperty.call(token.claims, 'admin')
+}
+
+const AuthContext = createContext({});
+
+const Auth = ({ children }) => {
+    const [user, setUser] = useState(null)
+    const [isLoading, setIsLoading] = useState(true)
+    const router = useRouter()
+
+    useEffect(() => {
+        const setPersistence = async () => {
+            await firebase.auth().setPersistence(firebase.auth.Auth.Persistence.SESSION)
```

Comment:
can we use LOCAL persistence instead? This makes it so the user doesn't have to re authenticate every time they open the cms.

```suggestion
            await firebase.auth().setPersistence(firebase.auth.Auth.Persistence.LOCAL)
```

Reply:
ya for sure! I think firebase auth persistence is defaulted to local for web apps, so I'm just gonna remove that effect hook.